spoc/io.py

```diff
@@ -73,12 +104,149 @@ def load_chromosome_sizes(path: str):
         )
 
     @staticmethod
-    def load_pixels(path: str):
-        """Loads pixels"""
-        # TODO:make this conform with new file format
-        raise NotImplementedError
+    def _load_metadata(path: str):
+        """Load metadata"""
+        metadata_path = Path(path) / "metadata.json"
+        if metadata_path.exists():
+            with open(metadata_path, "r", encoding="UTF-8") as f:
+                metadata = json.load(f)
+        else:
+            raise ValueError(f"Metadata file not found at {metadata_path}")
+        return metadata
+
+    @staticmethod
+    def list_pixels(path: str) -> List[PixelParameters]:
+        """List available pixels
+
+        Args:
+            path (str): Path to the pixel data.
+
+        Returns:
+            List[PixelParameters]: List of PixelParameters objects.
+        """
+        # read metadata.json
+        metadata = FileManager._load_metadata(path)
+        # instantiate pixel parameters
+        pixels = [PixelParameters(**params) for params in metadata.values()]
+        return pixels
+
+    @staticmethod
+    def list_contacts(path: str) -> List[ContactsParameters]:
+        """List available contacts"""
+        # read metadata.json
+        metadata = FileManager._load_metadata(path)
+        # instantiate pixel parameters
+        contacts = [ContactsParameters(**params) for params in metadata.values()]
+        return contacts
+
+    def load_pixels(
+        self, path: str, global_parameters: PixelParameters, load_dataframe: bool = True
+    ) -> Pixels:
+        """Loads specific pixels instance based on global parameters.
+        load_dataframe specifies whether the dataframe should be loaded, or whether pixels
+         should be instantiated based on the path alone.
+
+        Args:
+            path (str): Path to the pixel data.
+            global_parameters (PixelParameters): Global parameters.
+            load_dataframe (bool, optional): Whether to load the dataframe. Defaults to True.
+
+        Returns:
+            Pixels: Pixels object containing the pixel data.
+
+        """
+        metadata = self._load_metadata(path)
+        # find matching pixels
+        for pixel_path, value in metadata.items():
```

Comment:
I was thinking about why this search is O(N), when it could be O(1). Seems like we cannot hash all PixelParameters. But why? For example, for `metadata_combi` we could use `tuple` instead of `list`, since this collection seems to be immutable by design. Then `PixelParameters` can be hashed to be `dict` keys, so `_load_metadata` will return `Dict[PixelParameters: path]`.

Reply:
We change Metadata file keys are gloal parameters and values are filenames.

Reply:
Json dump can't encode parameters as keys as objects are not allowed as keys. We would need to serialize the object to a string and write a custom implemntation for that. Since this search operation is trivial (length of parameters is likely going to be small), I would vote for leaving it as is.

Reply:
Ok, let's leave it like this for now